consensus/hotstuff/committees/consensus_committee.go

```diff
@@ -79,39 +122,59 @@ var _ hotstuff.DynamicCommittee = (*Consensus)(nil)
 func NewConsensusCommittee(state protocol.State, me flow.Identifier) (*Consensus, error) {
 
 	com := &Consensus{
-		state:  state,
-		me:     me,
-		epochs: make(map[uint64]*staticEpochInfo),
+		state:                  state,
+		me:                     me,
+		epochs:                 make(map[uint64]*staticEpochInfo),
+		committedEpochsCh:      make(chan protocol.Epoch),
+		epochEmergencyFallback: make(chan struct{}),
 	}
 
+	com.Component = component.NewComponentManagerBuilder().
+		AddWorker(com.handleProtocolEvents).
+		Build()
+
 	final := state.Final()
 
-	// pre-compute leader selection for current epoch
-	current := final.Epochs().Current()
-	_, err := com.prepareEpoch(current)
+	// pre-compute leader selection for all presently relevant committed epochs
+	epochs := make([]protocol.Epoch, 0, 3)
+	// we always prepare the current epoch
+	epochs = append(epochs, final.Epochs().Current())
+
+	// we prepare the previous epoch, if one exists
+	exists, err := protocol.PreviousEpochExists(final)
 	if err != nil {
-		return nil, fmt.Errorf("could not add leader for current epoch: %w", err)
+		return nil, fmt.Errorf("could not check previous epoch exists: %w", err)
 	}
-
-	// Pre-compute leader selection for previous epoch, if it exists.
-	//
-	// This ensures we always know about leader selection for at least one full
-	// epoch into the past, ensuring we are able to not only determine the leader
-	// for block proposals we receive, but also adjudicate consensus-related
-	// challenges up to one epoch into the past.
-	previous := final.Epochs().Previous()
-	_, err = previous.Counter()
-	// if there is no previous epoch, return the committee as-is
-	if errors.Is(err, protocol.ErrNoPreviousEpoch) {
-		return com, nil
+	if exists {
+		epochs = append(epochs, final.Epochs().Previous())
 	}
+
+	// we prepare the next epoch, if it is committed
+	phase, err := final.Phase()
 	if err != nil {
-		return nil, fmt.Errorf("could not get previous epoch: %w", err)
+		return nil, fmt.Errorf("could not check epoch phase: %w", err)
+	}
+	if phase == flow.EpochPhaseCommitted {
```

Comment:
is this for an edge case where we have epoch committed but not yet finalized and we have reached the `EpochCommitmentDeadline`?